src/construction/construction.py

```diff
@@ -1,16 +1,243 @@
 """TODO."""
 
+import os
 import logging
 import pandas as pd
+from typing import Callable
+from datetime import datetime
 
 
 construction_logger = logging.getLogger(__name__)
 
 
-def run_construction(main_snapshot, updated_snapshot, config, write_csv, run_id):
-    """Define placeholder construction main function."""
+def run_construction(
+    main_snapshot: pd.DataFrame,
+    secondary_snapshot: pd.DataFrame,
+    config: dict,
+    check_file_exists: Callable,
+    write_csv: Callable,
+    read_csv: Callable,
+    run_id: int
+) -> pd.DataFrame:
+    """Run the construction module.
+
+    Args:
+        main_snapshot (pd.DataFrame): The staged and vaildated snapshot data.
+        secondary_snapshot (pd.DataFrame): The staged and validated updated
+            snapshot data.
+        config (dict): The pipeline configuration
+        check_file_exists (callable): Function to check if file exists. This will
+            be the hdfs or network version depending on settings.
+        write_csv (callable): Function to write to a csv file. This will be the
+            hdfs or network version depending on settings.
+        read_csv (callable): Function to read a csv file. This will be the hdfs or
+            network version depending on settings.
+        run_id (int): The run id for this run.
+    Returns:
+        constructed_df (pd.DataFrame): As main_snapshot but with records amended
+            and added from the construction files. amended
+    """
+    # Skip this module if the secondary snapshot isn't loaded
+    load_updated_snapshot = config["global"]["load_updated_snapshot"]
+    if load_updated_snapshot is False:
+        construction_logger.info("Skipping Construction...")
+        return main_snapshot
+
     # ! For now, we add the year column since neither file has it
     main_snapshot["year"] = 2022
```

Comment:
there's some code that does this in outputs/short_form.py - you could call the func or use this:
    df["period_year"] = df["period"].astype("str").str[:4]

(period_year is the col name for the short form outputs, would it work to use this in your code too, and we could just add it in during staging?